src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/DirectoryProvider.kt

```diff
@@ -27,37 +27,45 @@ class DirectoryProvider(private val appName: String) : IDirectoryProvider {
 
         val upperOS = os.toUpperCase()
 
-        val pathComponents = mutableListOf(appName)
+        val pathComponents = mutableListOf<String>()
+
+        when {
+            upperOS.contains("WIN") -> pathComponents.add(System.getenv("APPDATA"))
+            upperOS.contains("MAC") -> {
+                // use /Users/<user>/Library/Application Support/ for macOS
+                pathComponents.add(System.getProperty("user.home"))
+                pathComponents.add("Library")
+                pathComponents.add("Application Support")
+            }
+            upperOS.contains("LINUX") -> {
+                pathComponents.add(System.getProperty("user.home"))
+                pathComponents.add(".config")
+            }
+        }
+
+        pathComponents.add(appName)
 
         if (appendedPath.isNotEmpty()) pathComponents.add(appendedPath)
 
-        if (upperOS.contains("WIN")) {
-            // on windows use org.wycliffeassociates.otter.jvm.app data
-            pathComponents.add(0, System.getenv("APPDATA"))
-        } else if (upperOS.contains("MAC")) {
-            // use /Users/<user>/Library/Application Support/ for macOS
-            pathComponents.add(0, "Application Support")
-            pathComponents.add(0, "Library")
-            pathComponents.add(0, System.getProperty("user.home"))
-        } else if (upperOS.contains("LINUX")) {
-            pathComponents.add(0, ".config")
-            pathComponents.add(0, System.getProperty("user.home"))
-        }
         // create the directory if it does not exist
         val pathString = pathComponents.joinToString(separator)
         val file = File(pathString)
         file.mkdirs()
         return file
     }
 
-    override fun getProjectAudioDirectory(project: Collection, subcollections: List<Collection>): File {
-        // `subcollections` is a list of the collection hierarchy
-        // e.g., [ chapterCollection ]
-        var path = getUserDataDirectory("projects").resolve("${project.slug}_${project.id}")
-        // Build the folder structure
-        subcollections.forEach {
-            path = path.resolve(it.slug)
-        }
+    override fun getProjectAudioDirectory(
+            book: Collection,
+            chapterNumber: Int
+    ): File {
+        // <user data directory>/{language slug}/{rc slug}/{book slug}/{%02d, chapter number}/
+        val appendedPath = listOf(
+                book.resourceContainer?.language?.slug ?: "no_language",
+                book.resourceContainer?.identifier ?: "no_rc",
+                book.slug,
+                "%02d".format(chapterNumber)
```

Comment:
this is also affected by the %03 thing; perhaps chapter number should just be provided as a preformatted string?